volmdlr/edges.py

```diff
@@ -1379,29 +1393,37 @@ def rotation_inplace(self, center: volmdlr.Point2D, angle: float):
             point.rotation_inplace(center, angle)
 
     def line_intersections(self, line2d: Line2D):
-        polygon_points = self.discretization_points(number_points=100)
+        polygon_points = self.points
         list_intersections = []
         length = self.length()
         initial_abscissa = 0
         for points in zip(polygon_points[:-1], polygon_points[1:]):
             linesegment = LineSegment2D(points[0], points[1])
             intersections = linesegment.line_intersections(line2d)
-            initial_abscissa += linesegment.length()
-            if intersections:
+
+            if intersections and intersections[0] not in list_intersections:
+                abscissa = initial_abscissa + linesegment.abscissa(intersections[0])
                 if initial_abscissa < length * 0.1:
-                    list_abcissas = [initial_abscissa * n for n in
-                                     npy.linspace(0, 1, 100)]
+                    number_points = int(linesegment.length() / 1e-6)
+                    list_abcissas = list(
+                        n for n in npy.linspace(initial_abscissa, initial_abscissa + linesegment.length(),
+                                                number_points))
                 else:
-                    list_abcissas = [initial_abscissa * n for n in
-                                     npy.linspace(0.9, 1, 100)]
+                    resolution = 1e-6
+                    nb_pts = max(2, int(0.0001 / resolution))
+                    list_abcissas = list(new_abscissa for new_abscissa in npy.linspace(abscissa - 0.0001 / 2,
+                                                                                       abscissa + 0.0001 / 2, nb_pts))
                 distance = npy.inf
```

Comment:
maybe it's a good idea to put 0.0001 as a variable, so we can easily adjust if we need to debug someday.